sdk/keyvault/keyvault-certificates/tests/utils/recorder.ts

```diff
@@ -0,0 +1,434 @@
+    file.on("error", (err: any) => {
+      console.log(err);
+      throw err;
+    });
+
+    file.write(
+      importNock + "\n" + "module.exports.testInfo = " + JSON.stringify(this.uniqueTestInfo) + "\n"
+    );
+
+    for (const fixture of fixtures) {
+      // We're not matching query string parameters because they may contain sensitive information, and Nock does not allow us to customize it easily
+      const updatedFixture = fixture.replace(/\.query\(.*\)/, ".query(true)");
+      file.write(this.filterSecrets(updatedFixture) + "\n");
+    }
+
+    file.end();
+
+    nock.recorder.clear();
+    nock.restore();
+  }
+}
+
+class NiseRecorder extends Recorder {
+  private readonly sasQueryParameters = ["se", "sig", "sp", "spr", "srt", "ss", "st", "sv"];
+  private recordings: any[] = [];
+
+  constructor(testHierarchy: string, testTitle: string) {
+    super("browsers", testHierarchy, testTitle, "json");
+  }
+
+  // Inserts a request/response pair into the recordings array
+  private async recordRequest(request: any, data: any): Promise<void> {
+    const responseHeaders: any = {};
+    const responseHeadersPairs = request.getAllResponseHeaders().split("\r\n");
+    for (const pair of responseHeadersPairs) {
+      const [key, value] = pair.split(": ");
+      responseHeaders[key] = value;
+    }
+
+    // We're not storing SAS Query Parameters because they may contain sensitive information
+    // We're ignoring the "_" parameter as well because it's not being added by our code
+    // More info on "_": https://stackoverflow.com/questions/3687729/who-add-single-underscore-query-parameter
+    const parsedUrl = queryString.parseUrl(request.url);
+    const query: any = {};
+    for (const param in parsedUrl.query) {
+      if (!this.sasQueryParameters.includes(param) && param !== "_") {
+        query[param] = parsedUrl.query[param];
+      }
+    }
+
+    this.recordings.push({
+      method: request.method,
+      url: parsedUrl.url,
+      query: query,
+      requestBody: data instanceof Blob ? await blobToString(data) : data,
+      status: request.status,
+      response:
+        request.response instanceof Blob ? await blobToString(request.response) : request.response,
+      responseHeaders: responseHeaders
+    });
+  }
+
+  // Checks whether a recording matches a request or not (we're not matching request headers)
+  private matchRequest(recording: any, request: any): boolean {
+    // Every parameter in the recording must be present and have the same value in the request
+    for (const param in recording.query) {
+      if (recording.query[param] !== request.query[param]) {
+        return false;
+      }
+    }
+
+    // There shouldn't be parameters in the request that are not present in the recording (except for SAS Query Parameters and "_")
+    for (const param in request.query) {
+      if (
+        recording.query[param] === undefined &&
+        !this.sasQueryParameters.includes(param) &&
```

Comment:
Are sasQueryParameters part of the KV work too?

Reply:
This is a copy of the recorder.ts we have in keys and secrets. It will be replaced with a core recorder that Harsha is going to introduce soon.

For this specific change, I would need a bit of time to understand what's happening here, but it will be wasted since the core recorder is coming soon.